examples/scripts/TreeCompareAuto.C

```diff
@@ -0,0 +1,309 @@
+            okay = kFALSE;
+        }
+        if (0 == entries1 && 0 == entries2) {
+            outstream << "Both Histograms are empty." << std::endl;
+
+            hist1->Clear();
+            htemp->Clear();
+            //      numTestedLeaves++;
+            //      numEmptyLeaves++;
+            continue;
+        }
+
+        // When executing the draw command "Leaf1 - Leaf2" the subtraction is
+        // executed entry by entry. If the content of the class members are
+        // identical the result is a histogram with a delta function at 0
+        // If the content is differnt one gets a distribution which is
+        // detected.
+        outstream << "Comparing " << leafName << " and " << leafName1 << std::endl;
+        TString command = leafName + "-" + leafName1 + ">>hist(20, -10.,10.)";
+        tree1->Draw(command);
+        auto hist = (TH1F*)gPad->GetPrimitive("hist");
+        numTestedLeaves++;
+
+        // Check if the entries in the tree are identical
+        Bool_t leafIsIdentical = CheckEntriesIdentical(hist, outstream);
+
+        // If the entries are not identical check if the histograms are
+        // identical. This is the case if the entries in the tree are sorted
+        // differently
+        if (leafIsIdentical) {
+            numIdenticalEntries++;
+            outstream << "**********************" << std::endl;
+            hist1->Clear();
+            hist->Clear();
+            htemp->Clear();
+            continue;
+        } else {
+            outstream << "CHecking for identical histograms" << endl;
+            leafIsIdentical = CheckHistogramIdentical(htemp, hist1, outstream);
+        }
+
+        if (leafIsIdentical) {
+            numIdenticalHistograms++;
+            outstream << "**********************" << std::endl;
+            hist1->Clear();
+            hist->Clear();
+            htemp->Clear();
+            continue;
+        } else {
+            outstream << "CHecking Kolmogorov" << endl;
+            leafIsIdentical = CheckHistogramKolmogorov(htemp, hist1, outstream);
+        }
+
+        if (leafIsIdentical) {
+            numKolmogorovHistograms++;
+            outstream << "**********************" << std::endl;
+            hist1->Clear();
+            hist->Clear();
+            htemp->Clear();
+            continue;
+        } else {
+            outstream << "Data are differnt" << std::endl;
+            outstream << "**********************" << std::endl;
+            outstream << "Entries: " << hist->GetEntries() << std::endl;
+            outstream << "Mean: " << hist->GetMean() << std::endl;
+            outstream << "RMS: " << hist->GetRMS() << std::endl;
+            outstream << "Underflow: " << hist->GetBinContent(0) << std::endl;
+            outstream << "Overflow: " << hist->GetBinContent(hist->GetNbinsX() + 1) << std::endl;
+            outstream << "**********************" << std::endl;
+            okay = kFALSE;
+            numFailedLeaves++;
+        }
```

Comment:
```suggestion

        outstream << "Data are different" << std::endl;
        outstream << "**********************" << std::endl;
        outstream << "Entries: " << hist->GetEntries() << std::endl;
        outstream << "Mean: " << hist->GetMean() << std::endl;
        outstream << "RMS: " << hist->GetRMS() << std::endl;
        outstream << "Underflow: " << hist->GetBinContent(0) << std::endl;
        outstream << "Overflow: " << hist->GetBinContent(hist->GetNbinsX() + 1) << std::endl;
        outstream << "**********************" << std::endl;
        okay = kFALSE;
        numFailedLeaves++;
```

* no else block needed -> readability

Reply:
removed the else block